data/spells/scripts/support/magic_rope.lua

```diff
@@ -3,16 +3,18 @@ function onCastSpell(creature, variant)
 	position:sendMagicEffect(CONST_ME_POFF)
 
 	local tile = Tile(position)
-	if table.contains(ropeSpots, tile:getGround():getId()) or tile:getItemById(14435) then
-		tile = Tile(position:moveUpstairs())
-		if tile then
-			creature:teleportTo(position, false)
-			position:sendMagicEffect(CONST_ME_TELEPORT)
-		else
-			creature:sendCancelMessage(RETURNVALUE_NOTENOUGHROOM)
-		end
-	else
+	if not table.contains(ropeSpots, tile:getGround():getId()) or not tile:getItemById(14435) then
```

Comment:
The inverse of `table.contains(ropeSpots, tile:getGround():getId()) or tile:getItemById(14435)` is `not table.contains(ropeSpots, tile:getGround():getId()) and not tile:getItemById(14435))`

Reply:
Thanks Summ. Good eye! :)